src/translators/sql_translator.py

```diff
@@ -362,15 +362,12 @@ def _insert_entity_rows(self, table_name: str, col_names: List[str],
         stmt = f"insert into {table_name} ({col_list}) values ({placeholders})"
         try:
             start_time = datetime.now()
+            res_list = []
 
             for batch in to_insert_batches(rows):
                 res = self.cursor.executemany(stmt, batch)
-                # new version of crate does not bomb anymore when
-                # something goes wrong in multi entries
-                # simply it returns -2 for each row that have an issue
-                # TODO: improve error handling.
-                # using batches, we don't need to fail the whole set
-                # but only failing batches.
+                res_list.append(res)
+            for res in res_list:
```

Comment:
This approach will only work for Crate, not Timescale which I think throws an exception instead. In general, what to do here will depend on the database at hand, so we should actually push this code down to the translator subclasses.